src/components/RangeSelector.js

```diff
@@ -150,7 +144,7 @@ const RangeSelector = React.createClass({
       updatedState.dragging = 'Upper';
     }
 
-    this.setState(updatedState);
+    this.setState(updatedState, this._handleDragging(e));
```

Comment:
Since the `_handleDragging` access `state`, calling it as a chain will not guaranty it gets called with the updatedState. React provides a callback on the setState that will be triggered after the state is updated.